### ISSUE TITLE
KHR_draco_mesh_compression's bufferView decoded metadata can and does differ with the document's accessor values.

### ISSUE THREAD

**Issue by najadojo** [KhronosGroup/glTF#1342]
The Draco data stream contains vertex buffer metadata that mirrors that found in the accessor for the same buffer. KhronosGroup/glTF-Sample-Models#147 indicates that there are models where this metadata differs between the two sources data types. Additionally primitive vertex count can be different between the too. For example [BrainStem](https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/BrainStem/glTF-Draco) contains a primitive where the Draco vertex count is 806 and the accessor associated with that vertex attribute lists a count of 802. In this case there is no bufferView for the accessor, the accessor is simply giving incorrect information. 

The spec says that the bufferView be ignored when the draco extension is present for an attribute and the above presumes we also ignore the accessor to get expected result. Should we clarify that when no fallback is desired that the core attribute and its accessor not be present? This would eliminate the confusion above. It seems some clarifying language on the indices maybe necessary as well, the spec suggests to process the core but then read the data from draco. What does process mean here?

I'd also expect that when a fallback or mix and match attributes (ie some from draco some from core) is desired the data types and lengths match within a primitive. Ideally this is something that is validated by glTF-Validator.

**Comment by lexaknyazev** [comment]
Accessors configuration must match Draco output, so that engines could build VAOs before decompression ends. If some assets have that wrong, they must be fixed.

**Comment by najadojo** [comment]
Super! Can we assert this in the validator?

**Comment by FrankGalligan** [comment]
So when compressing with Draco (most of the options), this can change the number of vertices. Draco will remove vertices if they are not being used. The problem is if we want uncompressed fallback option we cannot just change the count to match what Draco compressed. So I made a change to the spec here:https://github.com/KhronosGroup/glTF/pull/1343
This change says to ignore accessor.count for compressed primitives.

I also made a change to gltf-pipeline here:https://github.com/FrankGalligan/gltf-pipeline/tree/add_uncompressed_fallback
that adds an option to output compressed and uncompressed meshes.
